apps/music/js/metadata/ogg.js

```diff
@@ -39,76 +44,127 @@ var OggMetadata = (function() {
    * @return {Promise} A Promise returning the parsed metadata object.
    */
   function parse(blobview, metadata) {
-    readIdentificationHeader(blobview);
-    return readCommentHeader(blobview, metadata);
+    var parser = new VorbisCommentParser(blobview, metadata);
+    parser.readIdentificationHeader(blobview);
+    return parser.readVorbisComment(blobview);
+  }
+
+  /**
+   * Constructor for the VorbisComment parser.
+   *
+   * @param {BlobView} blobview The audio file to parse.
+   * @param {Metadata} metadata The (partially filled-in) metadata object.
+   */
+  function VorbisCommentParser(blobview, metadata) {
+    this.file = blobview;
+    this.metadata = metadata;
+    this.continued = false;
+    this.seen_fields = {};
+    this.read_comments = 0;
+    return this;
   }
 
   /**
    * Read the identification header of an Ogg container. It's always the first
    * page.
    *
-   * @param {BlobView} page The audio file being parsed.
+   * @param {BlobView} blobview The audio file being parsed.
    */
-  function readIdentificationHeader(page) {
-    var header = readPageHeader(page);
+  VorbisCommentParser.prototype.readIdentificationHeader = function(blobview) {
```

Comment:
Do we need to pass the `blobview` to this function, given that you've already passed it to this object's constructor?